packages/commuter-client/src/actions/index.js

```diff
@@ -0,0 +1,44 @@
+import * as types from "../constants/ActionTypes";
+import { serverConfig } from "./../config";
+
+const jupyter = require("rx-jupyter");
+
+export const receiveNotebook = rawJson => ({
+  type: types.RECEIVE_NOTEBOOK,
+  rawJson: rawJson,
+  isFetching: false
+});
+
+export const requestNotebook = () => ({
+  type: types.REQUEST_NOTEBOOK,
+  isFetching: true
+});
+
+export const requestContents = () => ({
+  type: types.REQUEST_CONTENTS,
+  isFetching: true
+});
+
+export const receiveContents = contents => ({
+  type: types.RECEIVE_CONTENTS,
+  contents: contents,
+  isFetching: false
+});
+
+export const fetchNotebook = path => {
+  return dispatch => {
+    dispatch(requestNotebook());
+    return fetch(`/api/contents/${path}`)
+      .then(res => res.json())
+      .then(rawJson => dispatch(receiveNotebook(rawJson)));
+  };
+};
+
+export const fetchContents = path => {
+  return dispatch => {
+    dispatch(requestContents());
+    return jupyter.contents
+      .get(serverConfig, path)
+      .subscribe(res => dispatch(receiveContents(res.response.content)));
+  };
+};
```

Comment:
Later on we can make these epics so that we handle the separate cases around fetching the content, no worries for now though - this is a great start.